c++/src/kj/compat/http.c++

```diff
@@ -3677,6 +4110,19 @@ public:
               });
             }
 
+            kj::Maybe<CompressionParameters> compressionParameters;
+            if (settings.enableWebSocketCompression) {
+              KJ_IF_MAYBE(agreedParameters, responseHeaders.get(
+                  HttpHeaderId::SEC_WEBSOCKET_EXTENSIONS)) {
+                auto parseResult = _::tryParseExtensionAgreement(clientOffer,
+                    *agreedParameters);
+                if (parseResult.is<kj::Exception>()) {
+                  return settings.errorHandler.orDefault(*this).handleWebSocketProtocolError({
+                    502, "Bad Gateway", parseResult.get<kj::Exception>().getDescription(), nullptr});
+                }
+              }
```

Comment:
It's too late for fixups since this was merged. You could submit a bufix PR, or you could just include the fix as part of your next implementation PR. I assume this bug isn't actually affecting anyone since it's guarded by `enableWebSocketCompression` which no one should be enabling yet (since it's not actually implemented yet).